tpl-implementations/include/experimental/__p1673_bits/kokkos-kernels/blas1_swap_elements_kk.hpp

```diff
@@ -0,0 +1,94 @@
+
+#ifndef LINALG_TPLIMPLEMENTATIONS_INCLUDE_EXPERIMENTAL___P1673_BITS_KOKKOSKERNELS_SWAP_HPP_
+#define LINALG_TPLIMPLEMENTATIONS_INCLUDE_EXPERIMENTAL___P1673_BITS_KOKKOSKERNELS_SWAP_HPP_
+
+#include <utility>
+
+namespace KokkosKernelsSTD {
+
+// this is here until we can use kokkos 3.6 which has swap avail
+template <class T>
+KOKKOS_INLINE_FUNCTION void _my_tmp_swap(T& a, T& b) noexcept {
+  static_assert(
```

Comment:
Per P1673, we need Constraints, not Mandates.  That is, we would want to use concepts or `enable_if_t` to constrain the template parameters; we wouldn't want to use `static_assert`.

Reply:
yes totally right! Thanks. 
I need to get in the frame of mind of c++20 :) i am still mostly stuck at c++11/14 as you well know from sandia lagging

Reply:
Thanks @fnrizzi ! : - )  Constraints vs. Mandates isn't really a C++ >= 20 thing, but I totally understand needing to shift one's way of thinking.  I was lucky to be working for all but the last few months in a mostly C++20 code base.